ecosystem/sep-datastore.md

```diff
@@ -0,0 +1,174 @@
+## Preamble
+
+```
+SEP: To Be Assigned
+Title: Ledger Metadata Storage
+Author: Tamir Sen <@tamirms>
+Status: Draft
+Created: 2025-03-11
+Version: 0.1.0
+```
+
+## Simple Summary
+
+A standard for how [`LedgerCloseMeta`](https://github.com/stellar/stellar-xdr/blob/v22.0/Stellar-ledger.x#L539-L545)
+objects should be stored so that ledgers can be easily and efficiently ingested by downstream systems.
+
+## Dependencies
+
+None.
+
+## Motivation
+
+[galexie](https://github.com/stellar/go/tree/master/services/galexie) is a service which publishes
+[`LedgerCloseMeta`](https://github.com/stellar/stellar-xdr/blob/v22.0/Stellar-ledger.x#L539-L545) XDR objects to a GCS
+(Google Cloud Storage) bucket. However, the data format and layout of the XDR objects are not formally documented. This
+SEP aims to provide a comprehensive specification for storing LedgerCloseMeta objects, enabling third-party developers
+to build compatible data stores and clients for retrieving ledger metadata.
+
+## Specification
+
+The data store is a key-value store where:
+
+- **Keys** are strings following a specific hierarchical format.
+- **Values** are binary blobs representing compressed `LedgerCloseMetaBatch` XDR values.
+
+The key-value store must support:
+
+- Efficient point lookups for arbitrary keys.
+- Listing keys in lexicographic order, optionally filtered by a prefix.
+
+Examples of compatible key-value stores include Google Cloud Storage (GCS) and Amazon S3.
+
+---
+
+### Value Format
+
+Each value in the key-value store is the compressed binary encoding of the following XDR structure:
+
+```c++
+// Batch of ledgers along with their transaction metadata
+struct LedgerCloseMetaBatch
+{
+    // starting ledger sequence number in the batch
+    uint32 startSequence;
+
+    // ending ledger sequence number in the batch
+    uint32 endSequence;
+
+    // Ledger close meta for each ledger within the batch
+    LedgerCloseMeta ledgerCloseMetas<>;
+};
+```
+
+- A LedgerCloseMetaBatch represents a contiguous range of one or more consecutive ledgers.
+
+- All batches in a data store instance contain the same number of ledgers.
+
+---
+
+### Key Format
+
+Keys follow a hierarchical directory structure. The root directory is `/ledgers``, and subdirectories represent
+partitions. Each partition contains a fixed number of batches:
+
+```
+/ledgers/<partition>/<batch>.xdr
+```
+
+If the partition size is 1, the partition is omitted, resulting in:
+
+```
+/ledgers/<batch>.xdr
+```
+
+#### Partition Format:
+
+```go
+fmt.Sprintf("%08X--%d-%d/", math.MaxUint32-partitionStartLedgerSequence, partitionStartLedgerSequence, partitionEndLedgerSequence)
```

Comment:
this is fine, may be a stretch for people to visualize the output, for consideration, a mask that demonstrates same format with literal placeholder name/type:
`<partition_start_32bit_hex>--<partition_start_int>-<partition_end_int>`